fastlane/actions/test_sample_code.rb

```diff
@@ -59,8 +59,12 @@ def self.method_missing(method_sym, *arguments, &_block)
             config_item = available_options.find { |a| a.key == current_argument }
             UI.user_error!("Unknown parameter '#{current_argument}' for action '#{method_sym}'") if config_item.nil?
 
-            if config_item.data_type && !value.kind_of?(config_item.data_type) && !config_item.optional && !config_item.skip_type_validation
-              UI.user_error!("'#{current_argument}' value must be a #{config_item.data_type}! Found #{value.class} instead.")
+            unless config_item.optional
```

Comment:
skip_type_validation?

Wondering why we won't test it if the arg is optional?

Reply:
`skip_type_validation` happens inside of the `ensure_boolean_type_passes_validation` and the `ensure_generic_type_passes_validation`

And I don't SUPER know, TBH  😬 But it was in the original implementation and it was failing without it. There were some that were failing that were optional where sample was specifying an `ENV` which was causing it to fail so this fixed it 🙃